dGame/dComponents/RailActivatorComponent.cpp

```diff
@@ -13,12 +13,21 @@
 
 RailActivatorComponent::RailActivatorComponent(Entity* parent, int32_t componentID) : Component(parent) {
 	m_ComponentID = componentID;
-	const auto tableData = CDClientManager::Instance().GetTable<CDRailActivatorComponentTable>()->GetEntryByID(componentID);;
+}
 
-	m_Path = parent->GetVar<std::u16string>(u"rail_path");
-	m_PathDirection = parent->GetVar<bool>(u"rail_path_direction");
-	m_PathStart = parent->GetVar<uint32_t>(u"rail_path_start");
+void RailActivatorComponent::LoadConfigData() {
+	m_Path = m_ParentEntity->GetVar<std::u16string>(u"rail_path");
+	m_PathDirection = m_ParentEntity->GetVar<bool>(u"rail_path_direction");
+	m_PathStart = m_ParentEntity->GetVar<uint32_t>(u"rail_path_start");
+	m_DamageImmune = m_ParentEntity->GetVar<bool>(u"rail_activator_damage_immune");
+	m_NoAggro = m_ParentEntity->GetVar<bool>(u"rail_no_aggro");
+	m_NotifyArrived = m_ParentEntity->GetVar<bool>(u"rail_notify_activator_arrived");
+	m_ShowNameBillboard = m_ParentEntity->GetVar<bool>(u"rail_show_name_billboard");
```

Comment:
I've wondered, does this need to talk to the Player Forced movement component. that has two bools that are serialized ("is on rail" and "show billboard"), which this has vars for